app/terminal/terminal.tsx

```diff
@@ -72,7 +80,27 @@ export default class TerminalComponent extends React.Component<TerminalProps, Te
   }
 
   handleDownloadClicked() {
-    var element = document.createElement("a");
+    if (this.props.fullLogsFetcher) {
+      this.setState({ isLoadingFullLog: true });
+      const element = document.createElement("a");
```

Comment:
This can happen inside of the `then` block

Reply:
I left it out so that it could be referenced in the `finally` block, with the intention of putting the `remove` in the `finally` block being that we could still remove the element if we encountered an exception after it was added. But that's probably overly paranoid and we're not doing it below, so I'll swap it.